File: Napakalaki_Ruby/bad_consequence.rb
#encoding: utf-8

module NapakalakiGame
  class BadConsequence
    attr_reader :text, :level

    def initialize (aText, someLevels)
      @text = aText
      @level = someLevels
    end
    private_class_method :new

    def isEmpty
      raise NotImplementdError.new
    end

    def substractVisibleTreasure (t)
      raise NotImplementdError.new
    end

    def substractHiddenTreasure (t)
      raise NotImplementdError.new
    end

    def adjustToFitTreasureLists (v, h)
      raise NotImplementdError.new
    end

    def to_s
      "Text = #{@text}\nLevels = #{@level}\n"
    end
  end
end

File: Napakalaki_Java/src/NapakalakiGame/BadConsequence.java
    /*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author francisco
 */
public abstract class BadConsequence {
    protected String text;
    protected int levels;
    
    
    protected BadConsequence(String Text, int numLevels){
        text = Text;
        levels = numLevels;
    }
    
    public String getText(){
        return text;
    }
    
    public int getLevels(){
        return levels;
    }
    
    public abstract boolean isEmpty();
    
    public abstract void substractVisibleTreasure(Treasure t);
    
    public abstract void substractHiddenTreasure(Treasure t);
    
    public abstract BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h);
    
    @Override
    public String toString(){
        return "Text = " + text + "\nLevels = " + Integer.toString(levels) + "\n"; 
    }
 }
